packages/roosterjs-editor-plugins/test/TableResize/tableResizeTest.ts

```diff
@@ -0,0 +1,96 @@
+    const ADD_BUTTON = '</div>+</div>';
+    const TEST_ID = 'inserterTest';
+
+    beforeEach(() => {
+        editor = TestHelper.initEditor(TEST_ID);
+        plugin = new TableResize();
+        plugin.initialize(editor);
+    });
+
+    afterEach(() => {
+        editor.dispose();
+        plugin.dispose();
+        TestHelper.removeElement(TEST_ID);
+        document.body = document.createElement('body');
+    });
+
+    type Position = {
+        x: number;
+        y: number;
+    };
+
+    function initialize(): DOMRect {
+        editor.setContent(TABLE);
+        let tableRect: DOMRect = null;
+        editor.queryElements('table', table => {
+            const rect = table.getBoundingClientRect();
+            tableRect = rect;
+        });
+        return tableRect;
+    }
+
+    function runTest(mouseStart: Position, mouseEnd: Position, expectedResult: boolean) {
+        const editorDiv = editor.getDocument().getElementById(TEST_ID);
+
+        let mouseEvent = new MouseEvent('mousemove', {
+            clientX: mouseStart.x,
+            clientY: mouseStart.y,
+        });
+
+        editorDiv.dispatchEvent(mouseEvent);
+
+        const body = editor.getDocument().body;
+
+        expect(body.innerHTML.includes(ADD_BUTTON)).toBe(true);
+
+        mouseEvent = new MouseEvent('mousemove', {
+            clientX: mouseEnd.x,
+            clientY: mouseEnd.y,
+        });
+        editorDiv.dispatchEvent(mouseEvent);
+        expect(body.innerHTML.includes(ADD_BUTTON)).toBe(expectedResult);
+    }
+
+    it('removes the vertical inserter when moving the cursor out of the offset zone', () => {
+        const rect = initialize();
+        runTest({ x: rect.right, y: rect.top }, { x: rect.right / 2, y: rect.bottom }, false);
+    });
+
+    it('keeps the vertical inserter when moving the cursor inside the safe zone', () => {
+        const rect = initialize();
+        runTest(
+            { x: rect.right, y: rect.top },
+            { x: rect.right - insideTheOffset, y: rect.top },
+            true
+        );
+    });
+
+    it('removes the horizontal inserter when moving the cursor out of the offset zone', () => {
+        const rect = initialize();
+        runTest(
+            { x: rect.left, y: rect.bottom },
+            { x: (rect.right - rect.left) / 2, y: (rect.bottom - rect.top) / 2 },
+            false
+        );
+    });
+
+    it('keeps the horizontal inserter when moving the cursor out of the offset zone', () => {
+        const rect = initialize();
+        runTest(
+            { x: rect.left, y: rect.bottom },
+            { x: rect.left, y: rect.bottom - insideTheOffset },
```

Comment:
If my understanding is correct, this applies to the scenario where the culture language is left to right? but if it is right to left, the test for { x: rect.left, y: rect.bottom - insideTheOffset } will fail, and { x: rect.right, y: rect.bottom - insideTheOffset } will succeed.

I think we can add one more check here in this test case, something like:

runTest(
            { x: rect.left, y: rect.bottom },
            { x: isRTL ? rect.right : rect.left, y: rect.bottom - insideTheOffset },

you can use below to get isRTL

isRTL = getComputedStyle(this.editor.getDocument().body, 'direction') == 'rtl';

Reply:
It's correct this is only testing left to right scenarios, I will add more test to cover right to left culture language